projects/storefrontlib/cms-components/content/video/video.component.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AfterViewChecked,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import {
+  CmsBannerComponentMedia,
+  CmsResponsiveBannerComponentMedia,
+  CmsService,
+  CmsVideoComponent,
+  ContainerSizeOptions,
+  PageType,
+  SemanticPathService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { take, tap } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+import {
+  Media,
+  MediaContainer,
+} from '../../../shared/components/media/media.model';
+import { MediaService } from '../../../shared/components/media/media.service';
+
+@Component({
+  selector: 'cx-video',
+  templateUrl: './video.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VideoComponent implements AfterViewChecked, OnDestroy {
+  @HostBinding('class') styleClasses: string | undefined;
+  @ViewChild('videoPlayer') videoPlayer: ElementRef;
+
+  source: string | undefined;
+  thumbnail: Media | undefined;
+  routerLink: string | any[] | undefined;
+  autoPlay: boolean;
+  loop: boolean;
+  mute: string | undefined;
+  height: string;
+  protected subscriptions = new Subscription();
+
+  data$: Observable<CmsVideoComponent> = this.component.data$.pipe(
+    tap((data) => {
+      this.styleClasses = data.styleClasses;
+      this.setMedia(data.video, data.videoMedia);
+      this.setControls(data.autoPlay, data.loop, data.mute);
+      this.setVideoHeight(data.containerSize, data.videoContainerHeight);
+      this.setRouting(data);
+    })
+  );
+
+  constructor(
+    protected component: CmsComponentData<CmsVideoComponent>,
+    protected mediaService: MediaService,
+    protected urlService: SemanticPathService,
+    protected cmsService: CmsService,
+    protected cd: ChangeDetectorRef,
+    protected renderer: Renderer2
+  ) {}
+
+  ngAfterViewChecked(): void {
+    this.renderer.setStyle(
+      this.videoPlayer.nativeElement,
+      'height',
+      this.height
+    );
+  }
+
+  protected setMedia(
+    video?: CmsBannerComponentMedia,
+    media?: CmsBannerComponentMedia | CmsResponsiveBannerComponentMedia
+  ) {
+    if (video) {
+      this.source = this.mediaService.getMedia(video)?.src;
+    }
+
+    if (media) {
```

Comment:
need to check whether `data.containerBackground === ContainerBackgroundOptions.UPLOAD_RESPONSIVE_IMAGE` as well

Reply:
Done